src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -220,6 +230,28 @@ func (s *fileSystemSource) Read(
 	return results, nil
 }
 
+func (s *fileSystemSource) runMigrations(ctx context.Context, infoFilesByNamespace fs.InfoFilesByNamespace) {
+	// Only one migration for now, so just short circuit entirely if not enabled
+	if s.opts.MigrationOptions().TargetMigrationVersion() != migration.MigrationVersion_1_1 {
+		return
+	}
+
+	migrator, err := migrator.NewMigrator(migrator.NewOptions().
+		SetMigrationTaskFn(migration.MigrationTask).
+		SetInfoFilesByNamespace(infoFilesByNamespace).
+		SetMigrationOptions(s.opts.MigrationOptions()).
+		SetFilesystemOptions(s.fsopts).
+		SetInstrumentOptions(s.opts.InstrumentOptions()).
+		SetStorageOptions(s.opts.StorageOptions()))
+	if err != nil {
+		s.log.Error("error creating migrator. continuing bootstrap", zap.Error(err))
+	}
+
+	if err = migrator.Run(ctx); err != nil {
+		s.log.Error("error performing migrations. continuing bootstrap", zap.Error(err))
```

Comment:
What happens to this block if it errors? Does it just not get marked as bootstrapped by the fs bootstrapper and then get filled in by peer streaming?

Reply:
So my intention for any migration failures is that they should be logged as errors but not prevent processing of any filesets going forward. If we fail here (or anywhere, really), any filesets that should have been migrated but weren't will just continue to use the previous, unmigrated version.

Reply:
Hmm so with this particular migration it doesn't really matter since it's a bootstrap optimization, but migrations may potentially update data types or something else of that nature in future, where we'd want a more robust failure mechanism here... I think given the impact of the current migration it's probably ok to just log (also maybe add a metric to emit here just for tracking purposes)

Maybe would be a good compromise for kicking the can down the road to when we'd really need to flesh this out, while making sure that we'd remember to fix it then is to have the MigrationVersions we receive from `s.opts.MigrationOptions().TargetMigrationVersion()` have an `ErrorBehavior() (func(err error))` method that would be called here? Then we can explicitly say that e.g.  `MigrationVersion_1_1` error behavior would be a no-op

May be a bit overkill, wdyt?

Reply:
Ahh, that's not a bad idea. Sgtm.

Reply:
On second thought, decided to punt on this. There are two levels where you could see error handling being desired while migrating. One is at the individual task level and the second is at the migrator level (i.e. what's pointed out in this comment). We probably want to put a little more thought into the interplay between those two and provide flexibility for clients to decide if they want to fail at one level, both, or neither. I think I'd rather wait until we have a migration with a true error case and have us flesh it out then instead of putting in something now that may not entirely be what we want.